### PR TITLE
Support multiple Azure disks when creating machines

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area storage
/kind enhancement
/priority normal
/platform azure

**What this PR does / why we need it**:
After https://github.com/gardener/gardener/pull/1893 was merged quite some time ago, this PR now adds support for the Azure extension to handle multiple Azure disks for the worker machines.

**Which issue(s) this PR fixes**:
Ref https://github.com/gardener/machine-controller-manager/issues/462

**Special notes for your reviewer**:
* Thanks @guydaichs for your work and contributions!

**Release note**:

```noteworthy user
It is now possible to configure additional data volumes for the worker machines. Please consult [this documentation](https://github.com/gardener/gardener-extension-provider-azure/blob/master/docs/usage-as-end-user.md#workerconfig) for more information.
```

### CODE REVIEW

File: pkg/apis/azure/validation/shoot.go [hunk 1]
```diff
@@ -45,6 +45,17 @@ func ValidateWorkers(workers []core.Worker, zoned bool, fldPath *field.Path) fie
...
 
+		if length := len(worker.DataVolumes); length > 64 {
+			allErrs = append(allErrs, field.TooMany(fldPath.Index(i).Child("dataVolumes"), length, 64))
+		}
+		for j, volume := range worker.DataVolumes {
+			dataVolPath := fldPath.Index(i).Child("dataVolumes").Index(j)
+			if volume.Name == nil {
+				allErrs = append(allErrs, field.Required(dataVolPath.Child("name"), "must not be empty"))
+			}
+			allErrs = append(allErrs, validateVolume(volume.DeepCopy(), dataVolPath)...)
```
Comment (ialidzhikov) [inline pkg/apis/azure/validation/shoot.go:56]: Hmm, is `DeepCopy` required here as we do not perform any modifications?

File: pkg/apis/azure/validation/shoot.go [hunk 2]
```diff
@@ -45,6 +45,17 @@ func ValidateWorkers(workers []core.Worker, zoned bool, fldPath *field.Path) fie
 			allErrs = append(allErrs, validateVolume(worker.Volume, fldPath.Index(i).Child("volume"))...)
 		}
 
+		if length := len(worker.DataVolumes); length > 64 {
+			allErrs = append(allErrs, field.TooMany(fldPath.Index(i).Child("dataVolumes"), length, 64))
+		}
+		for j, volume := range worker.DataVolumes {
+			dataVolPath := fldPath.Index(i).Child("dataVolumes").Index(j)
+			if volume.Name == nil {
```
[the remark this answers is not in the document]
Comment (rfranzke) [reply pkg/apis/azure/validation/shoot.go:53]: Yes, I remove it. Actually, we should make this field a "required" one and remove the pointer type in g/g in the future.

[the remark this answers is not in the document]
Comment (jia-jerry) [reply pkg/apis/azure/validation/shoot.go]: But if the volume is used by system volume, it doesn't need a name. It seems better to be `optional`.

[the remark this answers is not in the document]
Comment (rfranzke) [reply pkg/apis/azure/validation/shoot.go]: Probably we can have two structures, one for the root disk, and one for optional data disks. Names would only be required for the data disks.

**Pull request merged**